BoardGameTrainer/Application/MainWindow.cs

```diff
@@ -70,20 +72,37 @@ public MainWindow(GameTrainerApplication application) : base(Gtk.WindowType.Topl
                 if (dialog.Run() == (int)ResponseType.Accept)
                 {
                     Assembly assembly = Assembly.LoadFile(dialog.Filename);
+                    bool isCorrectDll = true;
                     try
                     {
                         IGameManagerFactory factory = LoadGameFactory(assembly);
                         configWindow.UpdateGameFactoryDict(factory);
                     }
                     catch
                     {
+                        isCorrectDll= false;
                         MessageDialog error = new(dialog, DialogFlags.DestroyWithParent, MessageType.Error, 
                             ButtonsType.Ok, "Failed to add a new game");
                         error.Run();
                         error.Dispose();
                     }
+                    finally
```

Comment:
Po co to jest w bloku finally a nie po prostu sam if?

Reply:
w sumie chyba szczególnie dialog.Dispose() powinno być w finally, żeby na pewno się zadziało także, jak wystąpi wyjątek (inaczej chyba nie mamy tej pewności?).
Kopiowanie dllki nie musi być w finally, ale tak się to logicznie układa, jak ma byc przed dispose()